### DISCUSSION TITLE
Resource point of view / Laravel model style

### DISCUSSION THREAD

**Discussion by sxtnmedia** [saloonphp/saloon#255] [Ideas]
Hi everyone!

Sam, I'm impressed by your work and really loved this lib.

I'm currently developing a package based on the Saloon. Kind of a plugin, but a little bit more. It will give the possibility to use the Saloon from the perspective of a Resource in the style of Laravel Models.

```
$connector = new Gitlab()
$connector->issues()->find(1234); 
$item = Issue::find(1234);
$collection = Board::where('title', 'abcd')->get();
$response = Board::create(['title' => 'abcd']);

And even relationships:
$boardsCollection = Issue::find(1234)->boards;
$paginatedBoardsCollection = Issue::find(1234)->boards()->where('title' => 'abcd')->paginate();
```

Is there anyone interested in this kind of a package? :)

**Comment by Sammyjo20** [comment]
Hey, @sxtnmedia This sounds awesome! 😍 I'd love to see more, it's a yes from me. I've seen this concept written up before, but never implemented it for Saloon. 

Did you say this will be for Saloon? If so, always happy to give feedback :)

**Comment by sxtnmedia** [comment]
Yep, exclusively for Saloon ;) 

Here is the repo with PoC.
https://github.com/sxtnmedia/saloon-modelify

Earlier I made this from the scratch, writing whole SDK Builder but realised that I can get all goods from both worlds and not reinventing the wheel. So I adapted the basic functionality to the Saloon.

**Comment by sxtnmedia** [comment]
I don't know if someone other than me, needs this kind of approach to SDK building, but I just feel a great need to contribute something to the open source world, after all these years of using someone else's work ;)

**Comment by ahoiroman** [comment]
Sounds really cool and I definitely going to try this
